Frameworks/CoreGraphics/CGDataConsumer.mm

```diff
@@ -16,56 +16,119 @@
+                                                                           CFDataAppendBytes(static_cast<CFMutableDataRef>(info),
+                                                                                             (const UInt8*)data,
+                                                                                             size);
+                                                                           return size;
+                                                                       },
+
+                                                                       // CGDataConsumerReleaseInfoCallback
+                                                                       [](void* info) { CFRelease(static_cast<CFTypeRef>(info)); }
+};
+
+static const CGDataConsumerCallbacks sc_CGDataConsumerCFURLCallbacks{ // CGDataConsumerPutBytesCallback
+                                                                      [](void* info, const void* data, size_t size) -> size_t {
+                                                                          auto writeStream = woc::MakeAutoCF<CFWriteStreamRef>(
+                                                                              CFWriteStreamCreateWithFile(nullptr,
+                                                                                                          static_cast<CFURLRef>(info)));
+                                                                          if (!CFWriteStreamOpen(writeStream)) {
+                                                                              return 0;
+                                                                          }
+
+                                                                          CFIndex written =
+                                                                              CFWriteStreamWrite(writeStream, (const UInt8*)data, size);
+                                                                          CFWriteStreamClose(writeStream);
+                                                                          return (written < 0L) ? 0 : written;
+                                                                      },
+
+                                                                      // CGDataConsumerReleaseInfoCallback
+                                                                      [](void* info) { CFRelease(static_cast<CFTypeRef>(info)); }
+};
+
+struct __CGDataConsumer : CoreFoundation::CppBase<__CGDataConsumer> {
+    __CGDataConsumer(void* info, const CGDataConsumerCallbacks* callbacks) : m_info(info), m_callbacks(*callbacks) {
+    }
+    __CGDataConsumer(CFMutableDataRef data) : __CGDataConsumer((void*)CFRetain(data), &sc_CGDataConsumerCFDataCallbacks) {
+    }
+    __CGDataConsumer(CFURLRef url) : __CGDataConsumer((void*)CFRetain(url), &sc_CGDataConsumerCFURLCallbacks) {
+    }
+
+    size_t WriteData(const void* data, size_t size) {
```

Comment:
It's not able to be written to by end-users, only internally.  It seems to be a generic interface for a data sink, where CG wants to write some image data or something out to the user, where they can specify what URL or CFData or wherever the data will go.  I added this as the simplest "Write to Consumer" method, which AFAICT will only really be used by CGPDFContext and CGImageDestination if at all